src/dataflow/source/kafka.rs

```diff
@@ -901,21 +935,27 @@ fn downgrade_capability(
                     }
                 }
             }
-        }
 
-        //  Next, we determine the maximum timestamp that is fully closed. This corresponds to the minimum
-        //  timestamp across partitions.
-        let min = cp_info
-            .partition_metadata
-            .iter()
-            .map(|cons_info| cons_info.ts)
-            .min()
-            .expect("There should never be 0 partitions!");
-        // Downgrade capability to new minimum open timestamp (which corresponds to min + 1).
-        if changed && min > 0 {
-            dp_info.source_metrics.capability.set(min);
-            cap.downgrade(&(&min + 1));
-            cp_info.last_closed_ts = min;
+            //  Next, we determine the maximum timestamp that is fully closed. This corresponds to the minimum
+            //  timestamp
+            // across partitions.
+            let min = //match cp_info
+                cp_info.partition_metadata
+                .iter()
+                .map(|(_, cons_info)| cons_info.ts)
+                .min().unwrap();
+            /*{
```

Comment:
stray comment? Also, why was this commented out? (I don't immediately see why it isn't necessary to handle the no-partitions case)

Reply:
It was commented out because I initially did not create an entry for all partitions partition_metadata (this was wrong), so I removed it.  Sorry for forgetting to remove the comment!

We instantiate the system with "expected_partition_count=1" so we will never reach this line without there being at least one partition (nb: partition metadata, there won't necessarily be a consumer associated with it).

To ensure that workers with no partitions also downgrade their capability, I make every worker (independently of whether it owns the partition) downgrade its capability. This is not the most efficient (redundant work) but the best we can do (I think) without changing the consistency API